Add release notes for listdescriptors RPC (#21049)
bitcoin/bitcoin: doc/release-notes.md

Wallet
------

> - New `listdescriptors` RPC is available to inspect contents of a descriptor-enabled wallet.

Reviewer: ```suggestion
- A new `listdescriptors` RPC is available to inspect the contents of descriptor-enabled wallets.
```

  The RPC returns public versions of all imported descriptors, including their timestamp and flags.

>   For ranged descriptor it also returns range boundaries and the next index to generate addresses from.

Reviewer: ```suggestion
  For ranged descriptor it also returns range boundaries and the next index to generate addresses from. (#20226)
```

>   For ranged descriptor it also returns range boundaries and the next index to generate addresses from.

Reviewer: ```suggestion
  For ranged descriptors, it also returns the range boundaries and the next index to generate addresses from.
```
